app/models/match.rb

```diff
@@ -21,6 +21,7 @@ class MissingNamesError < StandardError; end
 
   validate :no_recent_match, on: :create
   before_create :save_user_info
+  after_create :notify_by_email, :notify_by_sms
```

Comment:
```suggestion
after_create_commit :notify_by_email, :notify_by_sms
```

Parfois il y a des race conditions avec Sidekiq qui trigger trop rapidement => attendre le `COMMIT` de la transaction SQL résout ce problème.

Reply:
+1 pour after_create_commit
Mais les race conditions viennent d'ailleurs. En fait deux matches pour les même user sont créés par deux jobs `SendCampaignJob` ici https://github.com/hostolab/covidliste/blob/master/app/jobs/send_campaign_job.rb#L32
J'avais rajouté une validation ici https://github.com/hostolab/covidliste/blob/master/app/models/match.rb#L22. mais ca n'a pas l'air d'avoir suffit